cmd/healthz/main.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gravitational/trace"
+
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/satellite/healthz/checks"
+	"github.com/gravitational/satellite/healthz/config"
+	"github.com/gravitational/satellite/healthz/handlers"
+	"github.com/gravitational/satellite/healthz/utils"
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+		fmt.Printf("ERROR: %v\n", err.Error())
+		os.Exit(255)
+	}
+}
+
+func run() error {
+	cfg := config.Config{}
+	config.ParseCLIFlags(&cfg)
+
+	if cfg.Debug {
+		trace.EnableDebug()
+		log.SetLevel(log.DebugLevel)
+	}
+	log.SetFormatter(&log.TextFormatter{})
+	log.SetOutput(os.Stderr)
+
+	log.Debug(trace.Errorf("starting using config: %#v", cfg))
+
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
```

Comment:
You should always release context resources, in simplest form with `defer cancel()` - regardless of whether you also invoke `cancel` explicitly (it is idempotent)